src/commands/trace/__tests__/trace.test.ts

```diff
@@ -36,5 +36,22 @@ describe('trace', () => {
         'circleci',
       ])
     })
+
+    test('should correctly detect the jenkins environment', () => {
+      process.env = {
+        DD_CUSTOM_TRACE_ID: 'abc',
```

Comment:
two things: 
* this will be injected by the jenkins plugin? How about running this in an "uninstrumented" jenkins instance? 
* this works differently from gitlab, right? In gitlab we do generate the id here, don't we?

Reply:
- Yes, it is generated by the Jenkins plugin. Uninstrumented Jenkins instances would generate a custom span which we would not be able to use in the backend. I'll add code to check for the presence of this variable and print a warning saying that you need to instrument the Jenkins instance if the variable does not exist.
- It works differently from circleci, yes. For circleci, the trace id is generated in the backend, so we don't need to report it in the custom span (we just need to report enough env var information to be able to re-calculate it in the backend too)

Reply:
I see, thanks for the explanation. We do generate the parent span id, don't we? And we pass it to the process we spawn:

https://github.com/DataDog/datadog-ci/blob/master/src/commands/trace/trace.ts#L50-L52 

Also, this [`process.env. DD_CUSTOM_PARENT_ID`](https://github.com/DataDog/datadog-ci/blob/master/src/commands/trace/trace.ts#L73) would be empty, wouldn't it? We only set it for the child process.

Reply:
In Jenkins we always generate a parent id for the same reason we generate a trace id: everything happens plugin-side

In CircleCI we do not generate a parent id if the parent is a job. If the parent is another custom id, since the custom ids are generated in datadog-ci, we populate the DD_CUSTOM_PARENT_ID variable so we can support customs spans having children.